src/AppBundle/v2/Registration/Assembler/LayDeputyshipDtoCollectionAssembler.php

```diff
@@ -0,0 +1,40 @@
+<?php
+
+namespace AppBundle\v2\Registration\Assembler;
+
+use AppBundle\v2\Registration\DTO\LayDeputyshipDtoCollection;
+
+class LayDeputyshipDtoCollectionAssembler
+{
+    /** @var CasRecToLayDeputyshipDtoAssembler */
+    private $layDeputyshipDtoAssembler;
+
+    /**
+     * @param LayDeputyshipDtoAssemblerInterface $layDeputyshipDtoAssembler
+     */
+    public function __construct(LayDeputyshipDtoAssemblerInterface $layDeputyshipDtoAssembler)
+    {
+        $this->layDeputyshipDtoAssembler = $layDeputyshipDtoAssembler;
+    }
+
+    /**
+     * @param array $data
+     * @return LayDeputyshipDtoCollection
+     */
+    public function assembleFromArray(array $data)
+    {
+        $collection = new LayDeputyshipDtoCollection();
+
+        foreach ($data as $uploadRow) {
+
+            if (!$this->layDeputyshipDtoAssembler->canAssemble($uploadRow)) {
+                throw new \InvalidArgumentException('Cannot assemble LayDeputyshipDto: Missing expected data');
+            }
```

Comment:
could move this to the assemble method

Reply:
I could but I felt that making `canAssemble` a part of the public interface would enforce any other assemblers (i.e a Sirius one) to implement the same check.

Reply:
On reflection, it does seem sensible to encapsulate it within the service, to prevent the call being bypassed. I'll update.